packages/special-pages/pages/new-tab/app/index.js

```diff
@@ -0,0 +1,89 @@
+ */
+export async function init (messaging, baseEnvironment) {
+    const init = await messaging.init()
+
+    if (!Array.isArray(init.widgets)) {
+        throw new Error('missing critical initialSetup.widgets array')
+    }
+    if (!Array.isArray(init.widgetConfig)) {
+        throw new Error('missing critical initialSetup.widgetConfig array')
+    }
+
+    // Create an instance of the global widget api
+    const widgetConfigAPI = new WidgetConfigAPI(messaging, init.widgetConfig)
+
+    // update the 'env' in case it was changed by native sides
+    const environment = baseEnvironment
+        .withEnv(init.env)
+        .withLocale(init.locale)
+        .withLocale(baseEnvironment.urlParams.get('locale'))
+        .withTextLength(baseEnvironment.urlParams.get('textLength'))
+        .withDisplay(baseEnvironment.urlParams.get('display'))
+
+    console.log('environment:', environment)
+    console.log('locale:', environment.locale)
+
+    const strings = environment.locale === 'en'
+        ? enStrings
+        : await fetch(`./locales/${environment.locale}/new-tab.json`)
+            .then(x => x.json())
+            .catch(e => {
+                console.error('Could not load locale', environment.locale, e)
+                return enStrings
+            })
+
+    const settings = new Settings({})
+        .withPlatformName(baseEnvironment.injectName)
+        .withPlatformName(init.platform?.name)
+        .withPlatformName(baseEnvironment.urlParams.get('platform'))
+
+    const didCatch = (error) => {
+        const message = error?.message || error?.error || 'unknown'
+        messaging.reportPageException({ message })
+    }
+
+    const root = document.querySelector('#app')
+    if (!root) throw new Error('could not render, root element missing')
+
+    render(
+        <EnvironmentProvider
+            debugState={environment.debugState}
+            injectName={environment.injectName}
+            willThrow={environment.willThrow}>
+            <ErrorBoundary didCatch={didCatch} fallback={<Fallback showDetails={environment.env === 'development'}/>}>
+                <UpdateEnvironment search={window.location.search}/>
+                <MessagingContext.Provider value={messaging}>
+                    <SettingsProvider settings={settings}>
+                        <TranslationProvider translationObject={strings} fallback={strings} textLength={environment.textLength}>
+                            <WidgetConfigProvider api={widgetConfigAPI} widgetConfig={init.widgetConfig} widgets={init.widgets}>
+                                <App>
+                                    <WidgetList />
+                                </App>
+                            </WidgetConfigProvider>
+                        </TranslationProvider>
+                    </SettingsProvider>
+                </MessagingContext.Provider>
+            </ErrorBoundary>
+        </EnvironmentProvider>
+        ,
+        root
+    )
+}
```

Comment:
This is pretty much the same setup as all pages, including environment stuff, app settings, translations etc